hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java

```diff
@@ -149,7 +151,8 @@ public void deleteResourceType(String typeId) {
         if (isEmpty(id)) {
             throw new IllegalArgumentException("Resource id must be not null");
         }
-        return Optional.ofNullable((Resource) resource.get(id));
+        IspnResource ispnResource = (IspnResource) resource.get(id);
+        return Optional.ofNullable(ispnResource == null ? null : ispnResource.getResource());
```

Comment:
not important, but that could be "return Optional.ofNullable(ispnResource).map(IspnResource::getResource)"

Reply:
+1